lib/stack_master/parameter_resolvers/env.rb

```diff
@@ -0,0 +1,18 @@
+module StackMaster
+  module ParameterResolvers
+    class Env < Resolver
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+      end
+
+      def resolve(value)
+        environment_variable = ENV[value]
+        raise ArgumentError, "The environment variable #{value} is not set" if environment_variable.to_s.empty?
```

Comment:
I think this should be `if environment_variable.nil?` because an empty parameter might be perfectly legitimate and `ENV` returns nil if the environment variable is _not set_

```
irb(main):007:0> ENV["FOO"]
=> nil
irb(main):008:0> exit
patrickrobinson:$ export FOO=""
patrickrobinson:$ irb
irb(main):001:0> ENV["FOO"]
=> ""
```

Reply:
Feels kind of odd. 
What's the context where `nil` is undesirable but `""` string might be legitimate? Do you have an example of how you might use this?

Reply:
`nil` is only possible if you do not set the Environment variable. Any empty string is only possible if you _explicitly_ set it to an empty string. In that case we should trust the user knows what they are doing and did this explicitly.

Reply:
@stevehodgkiss what are you're thoughts on this. Should it be allowed?

Reply:
I think an empty string is a valid input we should support. We could check `ENV.key?(value)` or `environment_variable.nil?` as @patrobinson mentioned.